christopher-holt/wk01/4-thurs/Objects_Homework/homework/homework.js

```diff
@@ -0,0 +1,253 @@
+
+        currentNumber = Number.parseInt(currentNumber); // turn the digit from a string into an integer
+
+        if(!Number.isInteger(currentNumber)) { //check that the digit is a number
+            return false;
+        }
+    }
+
+    // Number must be composed of at least two different digits (all the digits cannot be the same)
+    let obj = {};
+    for(let i = 0; i < creditCardNumber.length; i++) {
+        obj[creditCardNumber[i]] = true;
+    }
+    if(Object.keys(obj).length < 2) {
+        return false;
+    }
+
+    // The final digit must be even
+    if(creditCardNumber[creditCardNumber.length - 1] % 2 !== 0) {
+        return false
+    }
+
+    // The sum of all the digits must be greater than 16
+    let sum = 0;
+    for(let i = 0; i < creditCardNumber.length; i++) {
+        sum += Number(creditCardNumber[i]);
+        }
+        if(sum <= 16) {
+            return false;
+        } else {
+            return true;
+        }
+};
+
+console.log(validateCreditCard('9999777788880000')); // true
+console.log(validateCreditCard('6666666666661666')); // true
+console.log(validateCreditCard('a92332119c011112')); // false, has letters in it
+console.log(validateCreditCard('4444444444444444')); // false, only one type of number
+console.log(validateCreditCard('1111111111111110')); // false, sum less thn 16
+console.log(validateCreditCard('6666666666666661')); // false, ends with a odd number
+
+// JavaScript Bank
+// In this homework, you'll create a basic bank in Javascript. The bank has many accounts and the following capabilities that you need to write.
+
+// Bank
+// There is only one bank. This bank has an array of accounts. The bank needs a method that will return the total sum of money in the accounts. It also needs an addAccount method that will enroll a new account at the bank and add it to the array of accounts. There is no need to create additional functions of the bank to delete accounts, etc.
+
+// The bank has many accounts. Accounts should be objects that all share a set of common functionality.
+
+// Accounts
+// Accounts have a current balance and owner's name. You should be able to deposit or withdraw from an account to change the balance.
+
+// There is no need to write a user interface. Make sure functions return values -- you may also have your functions console.log() values to help you see your code working.
+
+// You should write a basic story through a series of JavaScript commands that shows that the methods do indeed work as expected: add some accounts, show the total balance, make some deposits and withdrawals, show the new total balance.
+
+// Tips
+// Don't overthink this. Shorter code is probably the answer.
+
+const Accounts = {
+    currentBalance: 1000,
+    ownersName: ['John'],
+    priceChange: []
+};
+
+const Bank = function() {
+    console.log(`Welcome ${Accounts.ownersName[0]} your Bank Balance is: $${Accounts.currentBalance}`);
+    Deposit(205);
```

Comment:
You could store the number 205 in a variable, then feed that variable both into your `Deposit()` function and `console.log()`.

Alternatively, you could have your `Deposit()` function return the `amount` and then do something like this when you call the function: `amountDeposited = Deposit(205);`
Then you could use the `amountDeposited` variable in the `console.log()`.

By the way, it's common practice in JS to name functions using _camelCase_ so ideally this function would be called `deposit()`.